RevenueCatUI/Templates/Components/Text/TextComponentViewModel.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  TextComponentView.swift
+//
+//  Created by Josh Holtz on 6/11/24.
+// swiftlint:disable missing_docs
+
+import Combine
+import Foundation
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+public class TextComponentViewModel: ObservableObject {
+
+    let locale: Locale
+    let localization: [String: String]
+    @Published private(set) var component: PaywallComponent.TextComponent
+
+    init(locale: Locale, localization: [String: String], component: PaywallComponent.TextComponent) {
+        self.locale = locale
+        self.localization = localization
+        self.component = component
+    }
+
+    var text: String {
+        // TODO: Replace variables like "{{ }}"
+        // TODO: Add logs?
+        if let textLid = component.textLid {
+            if let localizedText = localization[textLid] {
+                return localizedText
+            } else {
+                return component.text.value.first?.value as? String ?? "missing localized text for \(textLid)"
+            }
+        } else {
+            return component.text.value.first?.value as? String ?? "missing localized text"
+        }
```

Comment:
For anybody else reviewing, the way we get and validate locales are changing in the next PR